src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionList.java

```diff
@@ -262,7 +262,11 @@ public void onSuccess(List<StepExecution> result) {
                     }
                     LOGGER.log(Level.FINE, "Will resume {0}", result);
                     for (StepExecution se : result) {
-                        se.onResume();
+                        try {
+                            se.onResume();
+                        } catch (Throwable x) {
+                            se.getContext().onFailure(x);
```

Comment:
Should we return or exit the loop here?

Reply:
I do not think so. This step failed to resume, but others might resume fine, and there might be `retry` or other error handlers in place.

Also here I am just matching the established idiom of an `onResume` implementation doing a bunch of stuff all wrapped in a `try`-`catch` block.

Reply:
Oh, yeah for some reason I was thinking we were failing the Pipeline here, not the step.